### PR TITLE
Fix metric values going out of order on x axis

### LINKED ISSUE #45: Metrics chart wrongly connects points on X axis
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
- **MLflow installed from (source or binary)**: source
- **MLflow version (run ``python -c "from mlflow import version; print(version.VERSION)"``)**: 0.1.0
- **Python version**: 3.6.5
- **npm version (if running the dev UI)**: 6.0.1
- **Exact command to reproduce**: NA

### Describe the problem
In the UI at the metrics graph view, it looks like this for me:
image

The data points are logged at the end of an epoch using this Keras callback:
```
class LossHistory(Callback):
    def on_epoch_end(self, epoch, logs=None):
        log_metric("loss", logs.get('loss'))
        # in reality more metrics here
```
`logs.get('loss')` just returns a number. The connections of the dots in the interface should be connecting X=0 to X=1 to X=2 and so on. Why they are currently connected in this strange way, I have no clue.

### PR DESCRIPTION
**Pull request by mateiz**
This fixes #45, which happens because the JavaScript sorting code we had for combining metrics would sometimes reorder them based on the order of fields in some objects (especially when metrics had the same timestamp). Also related to #123.

### CODE REVIEW

File: mlflow/server/js/src/components/MetricView.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const COLORS = [
 
 class MetricView extends Component {
   static propTypes = {
-    title: PropTypes.string.isRequired,
+    title: PropTypes.element.isRequired,
```
Comment (mateiz) [inline mlflow/server/js/src/components/MetricView.js]: The wrong use of "string" here was causing a React error to be logged before

File: mlflow/server/js/src/components/MetricView.js [hunk 2]
```diff
@@ -63,6 +63,7 @@ class MetricView extends Component {
                             {this.props.runUuids.map((uuid, idx) => (
                                 <Line type="linear"
                                       dataKey={uuid}
+                                      key={uuid}
```
Comment (mateiz) [inline mlflow/server/js/src/components/MetricView.js]: This was also causing a React warning that the items in a list did not have distinct keys

File: mlflow/server/js/src/components/MetricView.js [hunk 3]
```diff
@@ -77,35 +78,25 @@ class MetricView extends Component {
...
-  const mergedMetricsByIdx = Utils.mergeRuns(runUuids, metricsByIdxList);
-
-  const metrics = [];
-  Object.keys(mergedMetricsByIdx).sort().forEach((idx) => {
-    let metric = { index: parseInt(idx, 10) };
-    runUuids.forEach((runUuid) => {
-      metric[runUuid] = mergedMetricsByIdx[idx][runUuid] || null;
-    });
-    metrics.push(metric);
+  let metrics = new Array(maxLength);
```
Comment (smurching) [inline mlflow/server/js/src/components/MetricView.js]: micronit: `const` instead of `let` here?

**Review by smurching — APPROVED** [review]
LGTM, manually confirmed it works by running the PyTorch example (`python example/tutorial/pytorch_tensorboard.py`) for different numbers of steps & viewing the UI afterwards.
